client/asset/btc/btc.go

```diff
@@ -493,14 +493,27 @@ func (btc *ExchangeWallet) Connect(ctx context.Context) (*sync.WaitGroup, error)
 	go func() {
 		defer wg.Done()
 		btc.run(ctx)
-		err := btc.wallet.LockUnspent(true, nil)
-		if err != nil {
-			btc.log.Errorf("failed to unlock %s outputs on shutdown: %v", btc.symbol, err)
-		}
+		btc.shutdown()
 	}()
 	return &wg, nil
 }
 
+func (btc *ExchangeWallet) shutdown() {
+	// Unlock any locked outputs.
+	err := btc.wallet.LockUnspent(true, nil)
+	if err != nil {
+		btc.log.Errorf("failed to unlock %s outputs on shutdown: %v", btc.symbol, err)
+	}
+	// Close all open channels for contract redemption searches
+	// to prevent leakages.
+	btc.findRedemptionMtx.Lock()
+	for contractOutpoint, req := range btc.findRedemptionQueue {
+		close(req.resultChan)
```

Comment:
Only sender should close.  If the find goroutines try to send or close the channel themselves, panic.  But in general it is not necessary to close a channel, at least not in terms of resource leaks.

Reply:
Ah, well you've got `findRedemptionMtx` guarding from that.  Still, there's no concern about leaking resources from unclosed channels.